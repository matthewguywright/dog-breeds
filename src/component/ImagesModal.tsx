import React, { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import {
  Typography,
  Modal,
  Box,
  Grid,
  ImageList,
  ImageListItem,
} from "@mui/material";
import shuffle from "lodash/shuffle";

function ImagesModal(props: any) {
  const breed = useSelector((state: any) => state.breed);

  const style = {
    position: "absolute" as "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    width: 800,
    bgcolor: "background.paper",
    border: "2px solid #000",
    boxShadow: 24,
    pt: 2,
    px: 4,
    pb: 3,
  };

  return (
    <Modal
      aria-labelledby="transition-modal-title"
      aria-describedby="transition-modal-description"
      open={props.isOpen}
      onClose={props.handleClose}
    >
      <Box sx={style}>
        <Typography id="modal-modal-title" variant="h6" component="h2">
          Gallery
        </Typography>
        {breed.imageList.length ? (
          <Grid container spacing={2}>
            <ImageList
              sx={{ width: 750, height: 450 }}
              cols={3}
              rowHeight={164}
            >
              {shuffle(breed.imageList).map((item: any, i: number) => (
                <ImageListItem key={item + i}>
                  <img
                    src={`${item}?w=164&h=164&fit=crop&auto=format`}
                    srcSet={`${item}?w=164&h=164&fit=crop&auto=format&dpr=2 2x`}
                    alt={item}
                    loading="lazy"
                  />
                </ImageListItem>
              ))}
            </ImageList>
          </Grid>
        ) : (
          <p>No images to display. Make a selection.</p>
        )}
      </Box>
    </Modal>
  );
}

export default ImagesModal;
