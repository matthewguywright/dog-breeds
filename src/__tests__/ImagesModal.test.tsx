import React from "react";
import ImagesModal from "../component/ImagesModal";
import { render, cleanup, screen } from "@testing-library/react";
import configureMockStore, { MockStore } from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureMockStore([]);
let store: MockStore;

const initialState = {
  imageList: ["test.jpg", "test2.jpg"] as any[],
};

afterEach(cleanup);

describe("components / app / ImagesModal", () => {
  it("should match the snapshot", () => {
    store = mockStore({ breed: initialState });
    const { asFragment } = render(
      <Provider store={store}>
        <ImagesModal />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should show an alert if there are no images", () => {
    store = mockStore({ breed: {
      imageList: []
    } });
    render(
      <Provider store={store}>
        <ImagesModal isOpen={true} handleClose={jest.fn} />
      </Provider>
    );

    const alertText = screen.getByText("No images to display. Please make a selection.");
    expect(alertText).toBeInTheDocument();
  });

  it("should show dog poster gallery title", () => {
    store = mockStore({ breed: {
      imageList: [
        'img1.jpg',
        'img2.jpg'
      ]
    } });
    render(
      <Provider store={store}>
        <ImagesModal isOpen={true} handleClose={jest.fn} />
      </Provider>
    );

    const titleText = screen.getByText(/Dog Poster Gallery/i);
    
    expect(titleText).toBeInTheDocument();
  });

  it("should show dog poster gallery image count", () => {
    store = mockStore({ breed: {
      imageList: [
        'img1.jpg',
        'img2.jpg'
      ]
    } });
    render(
      <Provider store={store}>
        <ImagesModal isOpen={true} handleClose={jest.fn} />
      </Provider>
    );

    const numberText = screen.getByText(/2 images/i);
    expect(numberText).toBeInTheDocument();
  });

  it("should contain 2 images", () => {
    store = mockStore({ breed: {
      imageList: [
        'img1.jpg',
        'img2.jpg'
      ]
    } });
    render(
      <Provider store={store}>
        <ImagesModal isOpen={true} handleClose={jest.fn} />
      </Provider>
    );

    const imageOne = screen.getByAltText("img1.jpg");
    const imageTwo = screen.getByAltText("img2.jpg");
    expect(imageOne).toBeInTheDocument();
    expect(imageTwo).toBeInTheDocument();
  });
});
