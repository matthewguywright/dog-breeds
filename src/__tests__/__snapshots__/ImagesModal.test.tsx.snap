// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components / app / ImagesModal should match the snapshot 1`] = `<DocumentFragment />`;
